### PR TITLE
Multiple new settings, app components import, multiple format exports, static folder, ...

### PR DESCRIPTION
**Pull request by fdoflorenzano**
This PR adds a bunch of options that enable more personalization apart from custom imports:
 
### Added

- Support to import custom `SideBar` React component that replaces sidebar in Mechanic UI. It's expected to be name-exported in  in new `app/` folder in mechanic project.
- New design function export `ExtraUi`, expected to be a React component that's added to the bottom section of sidebar, on top of default toggles and buttons.
- Support for SVG and PNG exports for SVG based engines like `engine-react` and `engine-svg`.
- New `hideFeedback` design function setting. When true, Mechanic's feedback button is hidden. Defaults to false.
- New `hideNavigation` design function setting. When true, the navigation input that let's users select a design function is hidden. Defaults to false.
- New `hidePresets` design function setting. When true, preset selection input is hidden. Defaults to false.
- New `hideScaleToFit` design function setting. When true, Scale to Fit toggle is hidden. Defaults to false.
- New `scaleToFit` design function setting. When false, Scale to Fit will be off initially. Defaults to true.
- New `hideAutoRefresh` design function setting. When true, Auto Refresh toggle is hidden. Defaults to false.
- New `autoRefresh` design function setting. When false, Auto Refresh will be off initially. Defaults to true.
- New `hideGenerate` design function setting. When true, Generate button is hidden. Defaults to false.
- New `showMultipleExports` design function setting. When false, single export button is shown. When true, two separate export buttons are shown: one for PNG export and another for SVG export. Defaults to false.
- New `ignoreStyles` design function setting. When true, CSS in iframe is injected into design function's SVG output. Defaults to false.
- Support to serve `static/` folder for all design functions. Allows to serve custom favicon.
- `_isPreview` value is passed to design function handler in `inputs` argument. Allows design function to adjust if the render is a preview or an export.

### CODE REVIEW

File: packages/core/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Added
+
+- Support to import custom `SideBar` component in `app/` folder that replaces sidebar in Mechanic UI.
+- New design function export `ExtraUi`, expected to be a React component that's added to the bottom section of sidebar, on top of default toggles and buttons.
+- Support for SVG and PNG exports for SVG based engines like `engine-react` and `engine-svg`.
+- New `hideFeedback` design function setting. When true, Mechanic's feedback button is hidden. Defaults to false.
+- New `hideNavigation` design function setting. When true, the navigation input that let's users select a design function is hidden. Defaults to false.
```
Comment (bravomartin) [inline packages/core/CHANGELOG.md:16]: lets

File: packages/core/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- New `hidePresets` design function setting. When true, preset selection input is hidden. Defaults to false.
```
Comment (bravomartin) [inline packages/core/CHANGELOG.md:17]: just to make sure... the preset's selector is still hidden unless there is a `presets` exports, right?

File: packages/core/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- New `hideScaleToFit` design function setting. When true, Scale to Fit toggle is hidden. Defaults to false.
+- New `scaleToFit` design function setting. When false, Scale to Fit will be off initially. Defaults to true.
```
Comment (bravomartin) [inline packages/core/CHANGELOG.md:19]: should this be `scaleToFitDefault` ?

File: packages/core/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- New `hideAutoRefresh` design function setting. When true, Auto Refresh toggle is hidden. Defaults to false.
+- New `autoRefresh` design function setting. When false, Auto Refresh will be off initially. Defaults to true.
```
Comment (bravomartin) [inline packages/core/CHANGELOG.md:21]: should this be `autoRefreshDefault` ?

File: packages/core/CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- New `hideGenerate` design function setting. When true, Generate button is hidden. Defaults to false.
+- New `showMultipleExports` design function setting. When false, single export button is shown. When true, two separate export buttons are shown: one for PNG export and another for SVG export. Defaults to false.
+- New `ignoreStyles` design function setting. When true, CSS in iframe is injected into design function's SVG output. Defaults to false.
+- Support to serve `static/` folder for all design functions.
+- `_isPreview` value is passed to design function handler in `inputs` argument.
```
Comment (bravomartin) [inline packages/core/CHANGELOG.md:26]: 💘

**Review by bravomartin — APPROVED** [review]
Beautiful! Just some minor comments, mostly formal. Take any that make sense to you.

Approved! 🥇

File: packages/core/CHANGELOG.md (hunk 5 above)
Comment (fdoflorenzano) [reply packages/core/CHANGELOG.md:26]: Not sure what this is about, but :)

File: packages/core/CHANGELOG.md (hunk 3 above)
Comment (fdoflorenzano) [reply packages/core/CHANGELOG.md]: Changed it to `initialScaleToFit`.

File: packages/core/CHANGELOG.md (hunk 2 above)
Comment (fdoflorenzano) [reply packages/core/CHANGELOG.md:17]: Yes!

File: packages/core/CHANGELOG.md (hunk 5 above)
Comment (bravomartin) [reply packages/core/CHANGELOG.md:28]: `_isPreview` ❤️‍🔥

**Pull request merged**